hydrascrape/action.py

```diff
@@ -21,11 +21,22 @@ def perror(txt, code=1):
     sys.exit(code)
 
 
-def nix_copy(cacheurl: str, paths: list[str], derivation: bool = False):
+def nix_copy(
+    cacheurl: str,
+    paths: list[str],
+    derivation: bool = False,
+    refresh: bool = False
+):
     """Copies stuff from cache"""
     if len(paths) < 1:
         return
 
+    if refresh:
+        refreshcmd = ["nix", "path-info", "--refresh",
+                      "--store", cacheurl]
+        refreshcmd.extend(paths)
+        subprocess.run(refreshcmd, stdout=subprocess.PIPE, check=False)
```

Comment:
LGTM!  btw maybe here specify the encoding explicitly would be good! it will help to avoid potential issues.